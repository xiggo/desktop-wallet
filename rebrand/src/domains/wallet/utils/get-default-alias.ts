import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let counter = sameCoinWallets.length;

	if (counter === 0) {
		counter = 1;
	}

	while (profile.wallets().findByAlias(makeAlias(counter))) {
		counter++;
	}

	return makeAlias(counter);
};
