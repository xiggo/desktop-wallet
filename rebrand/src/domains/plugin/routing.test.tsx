/* eslint-disable react/display-name */
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { RouterView } from "router";

import { render, screen } from "@/utils/testing-library";

describe("PluginRoutes", () => {
	it("should render", () => {
		const { asFragment } = render(
			<MemoryRouter>
				<RouterView routes={[{ component: () => <h1>PluginRoutes</h1>, path: "/" }]} />
			</MemoryRouter>,
		);

		expect(screen.getByText("PluginRoutes")).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});
});
