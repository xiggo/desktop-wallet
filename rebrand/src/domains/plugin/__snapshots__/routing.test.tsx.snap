// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginRoutes should render 1`] = `
<DocumentFragment>
  <div
    class="RouterView__Wrapper-sc-fll7t2-0"
    data-testid="RouterView__wrapper"
  >
    <h1>
      PluginRoutes
    </h1>
  </div>
</DocumentFragment>
`;
